src/methods.jl

```diff
@@ -175,26 +209,42 @@ FiniteDifferenceMethod:
   coefficients:          [0.08333333333333333, -0.6666666666666666, 0.0, 0.6666666666666666, -0.08333333333333333]
 
 julia> fdm(sin, 1, 1e-3)
-0.5403023058679624
+ 0.5403023058679624
 
 julia> fdm(sin, 1, 1e-3) - cos(1)  # Check the error.
 -1.7741363933510002e-13
 ```
 """
-@inline function (m::FiniteDifferenceMethod)(f::Function, x::Real, h::Real)
+function (m::FiniteDifferenceMethod{P,Q})(
+    f::TF,
+    x::Real,
+    step::Real
+) where {P,Q,TF<:Function}
     # Assume that converting to float is desired.
-    return _eval_method(m, f, float(x), h)
+    x = float(x)
+    fs = _evals(m, f, x, step)
+    return _eval_method(m, fs, x, step, m.coefs)
 end
-@inline function _eval_method(
+
+function _evals(
     m::FiniteDifferenceMethod,
-    f::Function,
+    f::TF,
     x::T,
-    h::Real
-) where T<:AbstractFloat
-    return sum(
-        i -> convert(T, m.coefs[i]) * f(T(x + h * m.grid[i])),
-        eachindex(m.grid)
-    ) / h^m.q
+    step::Real
+) where {TF<:Function,T<:AbstractFloat}
+    xs = x .+ T(step) .* m.grid
+    return f.(xs)
+end
+
+function _eval_method(
+    m::FiniteDifferenceMethod{P,Q},
+    fs::SVector{P,TF},
+    x::T,
+    step::Real,
+    coefs::SVector{P,Float64}
+) where {P,Q,TF,T<:AbstractFloat}
+    coefs = T.(coefs)
+    return sum(fs .* coefs) ./ T(step)^Q
```

Comment:
```suggestion
    return sum(fs .* T.(coefs)) ./ T(step)^Q
```
?

Reply:
For some reason this substitution causes the method to allocate. I'm not 100% sure why. I've added a comment to explain the current formatting of the code.

Reply:
Hmm yeah, odd. This comment definitely suffices. Thanks!

Reply:
I have a MWE
https://github.com/JuliaLang/julia/issues/39151